alexandria/core/api.py

```diff
@@ -0,0 +1,66 @@
+from django.conf import settings
+from django.core.exceptions import ValidationError
+
+from alexandria.core import models
+
+
+def create_document_file(user, group, document_attributes, file_attributes):
+    """
+    Create a document and file with the given attributes.
+
+    This function eases the creation of documents and files by automatically setting important fields.
+    Uses ``create_file`` to create the file.
+
+    :param user: The user creating the document and file.
+    :param group: The group creating the document and file.
+    :param document_attributes: A dictionary containing the fields for the document. (required: category, title)
+    :param file_attributes: A dictionary containing the fields for the file. (required: name, content, mime_type, size)
+    :return: A tuple containing the created document and file.
+
+    """
+    document = models.Document.objects.create(
+        created_by_user=user,
+        created_by_group=group,
+        modified_by_user=user,
+        modified_by_group=group,
+        **document_attributes,
+    )
+    file = create_file(document, user, group, file_attributes)
+
+    return document, file
+
+
+def create_file(document, user, group, attributes):
```

Comment:
I'd like to see `**kwargs` instead of passing around dicts, at least for the optional params. The required ones should be regular args without defaults